Chapter1/Tests/CalculatorTests/DivisionTests.swift

```diff
@@ -0,0 +1,70 @@
+import XCTest
+@testable import Calculator
+
+extension CalculatorTests {
+  func testDivision_whenDividingBySameNumber_returnsOne() throws {
+    let first = 5
+    let second = 5
+    
+    let result = try Calculator.division(first, second)
+    
+    XCTAssertEqual(result, 1)
+  }
+  
+  func testDivision_whenDividingByZero_returnsNil() throws {
+    var thrownError: Error?
+    let first = 3
+    let second = 0
+        
+    XCTAssertThrowsError(try Calculator.division(first, second)) {
+      thrownError = $0
+    }
+    XCTAssertTrue(thrownError is Calculator.CalculatorError)
+    XCTAssertEqual(thrownError as? Calculator.CalculatorError, .divisionByZero)
+  }
```

Comment:
This is a good approach. Another alternative is to assert the error directly in the closure:

```swift
XCTAssertThrowsError(try Calculator.division(first, second)) {
  guard let calcError = try? XCTUnwrap($0 as Calculator.CalculatorError) else {
    XCTFail("Error is not a Calculator error but a \(type(of: $0))")
    return
  }
  XCTAssertEqual(calcError, .divisionByZero
}
```